### PR TITLE
FOUR-8393: Elements are moved out of the Pool container

### PR DESCRIPTION
**Pull request by rodriquelca**
## Issue & Reproduction Steps

Expected behavior: 
The elements should not be moved out of the Pool container

Actual behavior: 
The elements copied are moved  out of the Pool container
## Solution
- saveState logic was moved to the selection and drag side
- flows into container will be updated after the stopDrag

https://github.com/ProcessMaker/modeler/assets/1401911/518f1eee-830a-4383-8cd1-aa320660d82b

https://github.com/ProcessMaker/modeler/assets/1401911/95964ec0-f610-4ace-8343-dd46e69bc6f8

## How to Test
Test the steps above
- npm install
- npm run serve
- check the steps in the related tickets

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-8393
- https://processmaker.atlassian.net/browse/FOUR-8651

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

### CODE REVIEW

**Review by josechirivella — APPROVED** [review]

**Review by TinCodes — APPROVED** [review]

**Review by caleeli — APPROVED** [review]
Excelente change!

**Review by ana-mauricio — CHANGES_REQUESTED** [review]
The following observations were found in thi PR:
**Regression tickets:**

1. the elements go out of the pool with two pools https://processmaker.atlassian.net/browse/FOUR-8677

**Reopen tickets**
The following ticket was replicated again in this pool

1. https://processmaker.atlassian.net/browse/FOUR-7809

**New ticket no regressions**
The following tickets are related to selection of flows with boundary events were identified during the revision

1. https://processmaker.atlassian.net/browse/FOUR-8679
2. https://processmaker.atlassian.net/browse/FOUR-8680
@rodriquelca @boliviacoca

**Comment by ana-mauricio** [comment]
The PR is approved the following errors were fixed 
 - https://processmaker.atlassian.net/browse/FOUR-8677 Fixed
 - https://processmaker.atlassian.net/browse/FOUR-7809 Fixed
 - https://processmaker.atlassian.net/browse/FOUR-8689 Fixed
 - https://processmaker.atlassian.net/browse/FOUR-8675 Fixed
 - https://processmaker.atlassian.net/browse/FOUR-8674 Fixed
 - https://processmaker.atlassian.net/browse/FOUR-8393 Fixed
 - https://processmaker.atlassian.net/browse/FOUR-8651 Fixed
@rodriquelca @boliviacoca

**Review by ana-mauricio — APPROVED** [review]

File: src/components/crown/crownConfig/crownConfig.vue [hunk 1]
```diff
@@ -222,8 +222,6 @@ export default {
       if (!store.getters.allowSavingElementPosition) {
         return;
       }
-
-      this.$emit('save-state');
```
Comment (danloa) [inline src/components/crown/crownConfig/crownConfig.vue:226]: is it necessary to remove this $emit? It looks like the undo for actions started from the crow can be affected. Can you explain?

File: src/components/nodes/genericFlow/SequenceFlow.js [hunk 2]
```diff
@@ -6,9 +6,10 @@ import DataAssociation from '@/components/nodes/genericFlow/DataAssociation';
 
 export default class SequenceFlow extends Flow {
   static isValid({ sourceShape, targetShape, targetConfig }) {
+ 
     const targetNode = get(targetShape, 'component.node');
     const sourceNode = get(sourceShape, 'component.node');
-
+
```
Comment (TinCodes) [inline src/components/nodes/genericFlow/SequenceFlow.js:12]: Remove whitespace

File: src/components/nodes/boundaryEvent/boundaryEvent.vue [hunk 3]
```diff
@@ -174,8 +174,6 @@ export default {
       this.shape.listenToOnce(this.paper, 'cell:pointerup blank:pointerup', () => {
         this.moveBoundaryEventIfOverTask();
         this.resetInvalidTarget();
-        this.$emit('save-state');
```
Comment (TinCodes) [inline src/components/nodes/boundaryEvent/boundaryEvent.vue:177]: Does this cause the undo/redo to ignore when a boundary event is created and attached?

**Review by danloa — APPROVED** [review]

**Review by boliviacoca — APPROVED** [review]

**Comment by caleeli** [comment]
There is still an issue with the anchor points pools and copy paste

[ScreenMoveAnchorPointsCopyPaste.webm](https://github.com/ProcessMaker/modeler/assets/8028650/001ce791-a6b0-438c-b4e4-a95cd45cf7b4)

**Comment by rodriquelca** [comment]
added a validation to prevent waypoint movements in copy/paste actions

File: src/components/modeler/Selection.vue [hunk 4]
```diff
@@ -589,6 +696,21 @@ export default {
       let selectedArea = g.rect(f.x, f.y, width, height);
       return this.getElementsInSelectedArea(selectedArea, { strict: false });
     },
+    getPool(elements){
+      const { paper } = this.paperManager;
+      let pool = null;
+      if (elements.length > 0) {
```
Comment (josechirivella) [inline src/components/modeler/Selection.vue:702]:
```suggestion
      if (elements && elements.length > 0) {
```

**Review by josechirivella — APPROVED** [review]

**Pull request merged**